src/feature/Registration/index.jsx

```diff
@@ -35,23 +76,62 @@ const Registration = () => {
     setConfirmPasswordValid(validateConfirmPassword(password, value));
   };
 
-  const handleClick = (event) => {
-    event.preventDefault();
-    if (!validateEmail(email)) {
-      alert("Please enter a valid email address.");
+  const handleRegistrationEvent = async (e) => {
+    e.preventDefault();
+
+    if (!firstName || !lastName || !email || !password || !confirmPassword) {
+      setIsEmptyInput(true);
       return;
     }
 
-    if (!validatePassword(password)) {
-      alert("Please enter a valid password (at least 6 characters long).");
+    if (!validateConfirmPassword(password, confirmPassword)) {
+      setRegistrationFailed(true);
       return;
     }
 
-    if (!validateConfirmPassword(password, confirmPassword)) {
-      alert("Passwords do not match. Please re-enter.");
+    if (!emailValid || !passwordValid || !confirmPasswordValid) {
+      setRegistrationFailed(true);
       return;
     }
-    navigate("/dashboard");
+
+    setIsRegistering(true);
+
+    const userCredentials = {
+      firstName,
+      lastName,
+      email,
+      password,
+    };
+
+    try {
+      const response = await fetch(
+        "https://afrihub-backend.onrender.com/api/auth/signup",
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(userCredentials),
+        }
+      );
+
+      if (response) {
+        const data = await response.json();
```

Comment:
Start adopting custom hooks for http requests and services.
Follow this tutorial to refactor this into a custom hook `useFetch`.

https://blog.openreplay.com/building-a-custom-fetch-hook-in-react/#:~:text=Building%20the%20custom%20Fetch%20Hook&text=You%20can%20name%20the%20hook,are%20fetching%20the%20data%20from.